uber/models.py

```diff
@@ -470,6 +470,73 @@ def jobs_for_signups(self):
                     restricted_hours.add(frozenset(job.hours))
             return [job.to_dict(fields) for job in jobs if job.restricted or frozenset(job.hours) not in restricted_hours]
 
+        def available_badge_types(self):
+            """
+            There are several contexts where we want to display different badge types to select:
+            - A new attendee registering for the event
+            - An attendee claiming a blank badge in an existing group
+            - An existing attendee editing their registration
+
+            This function uses the parameters in the session to figure out the context and return the correct
+            badge types to display for an attendee.
+            """
+            params = cherrypy.lib.httputil.parse_query_string(cherrypy.request.query_string)
+            group_id = params['group_id'] if 'group_id' in params else None
```

Comment:
These can probably be combined so there's fewer interim steps - I just wanted to push out a working version of the code, first. :)